<launch>  
  <!-- must be passed in when calling this launch file -->
  <arg name="gdb" />
  
  <!-- translate keyboard input to TOWR commands -->
  <node 
    name="user_interface" 
    pkg="towr_ros" 
    type="towr_user_interface" 
    output="screen"
    launch-prefix="xterm -e">  <!-- open in new window -->
  </node>
  
  <!-- draw the different terrains in RVIZ -->
  <node 
    name="rviz_terrain_publisher" 
    pkg="towr_ros" 
    type="rviz_terrain_publisher" 
    output="screen"
    launch-prefix="$(arg gdb)"
    >
  </node>
  
  <!--  visualize the goal pose in RVIZ -->
  <node 
    name="goal_pose_publisher" 
    pkg="towr_ros" 
    type="goal_pose_publisher" 
    output="screen"
    launch-prefix="$(arg gdb)"
    >
  </node>

  <!-- start the actual NLP solver using TOWR -->
  <node 
    name="towr_ros_app" 
    pkg="towr_ros" 
    type="towr_ros_app" 
    output="screen"
    launch-prefix="$(arg gdb)"
    >
  </node>
    
</launch>
